src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchCasesOperation.java

```diff
@@ -0,0 +1,170 @@
+package uk.gov.hmcts.ccd.domain.service.aggregated;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CommonField;
+import uk.gov.hmcts.ccd.domain.model.definition.*;
+import uk.gov.hmcts.ccd.domain.model.search.CaseSearchResult;
+import uk.gov.hmcts.ccd.domain.model.search.UseCase;
+import uk.gov.hmcts.ccd.domain.model.search.elasticsearch.*;
+import uk.gov.hmcts.ccd.endpoint.exceptions.BadRequestException;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+
+@Service
+public class MergeDataToSearchCasesOperation {
```

Comment:
temporary: I would not model this as an operation with an execute method. Because by convention in Data Store Operation classes correspond to business operations like create case, view case. Even thought I can see we are introducing them for other other things recently